code/go/0chain.net/smartcontract/faucetsc/models.go

```diff
@@ -113,12 +114,39 @@ func (gn *GlobalNode) updateConfig(fields map[string]string) error {
 			gn.OwnerId = value
 
 		default:
+			if strings.HasPrefix(key, Settings[Cost]) {
```

Comment:
Do the strings.HasPrefix in the `gn.setCostValue` to make this easier to read.

Reply:
That will not work as I need the return statement to work only if it has the prefix, otherwise I need it to return the error from the next line.

Reply:
The way I suggested was do it like this:
```go
func (gn *GlobalNode) setCostValue(key, value string) error {
    if !strings.HasPrefix(key, Settings[Cost]) {
      return fmt.Errorf("key %s not recognised as setting", key)
  }
  // do the same as before
}

func updateConfig(..) error {
...
  switch {
    ...
    default:
      return gn.setCostValue()
  }
...
}
```

The reasons I suggest to move the prefix checking to `gn.setCostValue` are:
a) if the `gn.setCostValue()` is called more than one place, you will need to do the prefix checking in all the places that it is called, although it is only used in `updateConfig` now.
b) the code will be more cleaner
c) The `setCostValue()` actually should do a prefix checking itself to avoid unnecessary work instead of relying on the checking by others.

Reply:
got it, will have it changed.

Reply:
done